hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/management/GetClusterMetadataMessageTask.java

```diff
@@ -57,7 +58,8 @@ protected ClientMessage encodeResponse(Object response) {
                 metadata.getCurrentState().getId(),
                 metadata.getMemberVersion(),
                 null,
-                metadata.getClusterTime());
+                metadata.getClusterTime(),
+                metadata.getClusterId());
```

Comment:
if `isClusterIdExists=false`, i.e. older server versions, then the cluster id field is set to invalid number probably all zeros. Do you consider utilizing this field?

Reply:
No, we are encoding the response here, so we don't have to do any compatibility checks. They need to do that on the MC repo, where they decode the response

Reply:
Would it be null in that situation if we change this field to nullable? Then maybe we can more easily tackle it on the MC side.

Reply:
UUID is nullable by default, so that won't help you.

The logic around this new parameter will be
-> `isClusterIdExists=true`, `clusterId=a valid UUID or null ` (the method you call to get the cluster id is defined as nullable, so it might be null even if it exists. See `ClusterService#getClusterId` method signature) for 5.2 and newer members
-> `isClusterIdExists=false`, `clusterId=null` for older members (5.1 and below)

You need to handle this in the MC side according to this logic

Reply:
thanks for the explanation